pulsar-broker/src/test/java/org/apache/pulsar/broker/auth/MockedPulsarServiceBaseTest.java

```diff
@@ -717,5 +720,16 @@ protected void sleepSeconds(int seconds){
         }
     }
 
+    public static void reconnectAllConnections(PulsarClientImpl c) throws Exception {
+        ConnectionPool pool = c.getCnxPool();
+        Method closeAllConnections = ConnectionPool.class.getDeclaredMethod("closeAllConnections", new Class[]{});
+        closeAllConnections.setAccessible(true);
+        closeAllConnections.invoke(pool, new Object[]{});
+    }
+
+    public void reconnectAllConnections() throws Exception {
```

Comment:
This breaks Pulsar SQL / Trino tests in branch-3.1 and before. I created #21976 to address the problem.